bittensor/utils/registration.py

```diff
@@ -19,62 +23,77 @@
 from .formatting import get_human_readable, millify
 from ._register_cuda import solve_cuda
 
-try:
-    import torch
-except ImportError:
-    torch = None
-
 
 def use_torch() -> bool:
+    """Force the use of torch over numpy for certain operations."""
     return True if os.getenv("USE_TORCH") == "1" else False
 
 
-class Torch:
-    def __init__(self):
-        self._transformed = False
+def legacy_torch_api_compat(func):
+    """Decorator to convert numpy arrays to torch tensors before passing them to the function.
```

Comment:
I really like this approach. I would recommend updating this line, as it reads as the opposite of what it's saying. Other than that, this seems great.